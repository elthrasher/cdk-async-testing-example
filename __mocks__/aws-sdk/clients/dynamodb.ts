// See https://dev.to/elthrasher/mocking-aws-with-jest-and-typescript-199i
import { awsSdkPromiseResponse } from './awsSdkPromiseResponse';

export const deleteFn = jest.fn().mockImplementation(() => ({ promise: awsSdkPromiseResponse }));

export const getFn = jest.fn().mockImplementation(() => ({ promise: awsSdkPromiseResponse }));

const queryFn = jest.fn().mockImplementation(() => ({ promise: awsSdkPromiseResponse }));

export const putFn = jest.fn().mockImplementation(() => ({ promise: awsSdkPromiseResponse }));

export const updateFn = jest.fn().mockImplementation(() => ({ promise: awsSdkPromiseResponse }));

export class DocumentClient {
  delete = deleteFn;
  get = getFn;
  put = putFn;
  query = queryFn;
  update = updateFn;
  options = {};
}
