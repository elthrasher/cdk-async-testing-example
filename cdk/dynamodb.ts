import { AttributeType, Table } from '@aws-cdk/aws-dynamodb';
import { RemovalPolicy, Stack } from '@aws-cdk/core';

export const getTable = (stack: Stack): Table => {
  return new Table(stack, 'PaymentsTable', {
    partitionKey: { name: 'pk', type: AttributeType.STRING },
    removalPolicy: RemovalPolicy.DESTROY,
    tableName: 'payments',
  });
};
