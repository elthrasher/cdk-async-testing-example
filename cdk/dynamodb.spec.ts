import { SynthUtils } from '@aws-cdk/assert';
import { App, Stack } from '@aws-cdk/core';

import { getTable } from './dynamodb';

describe('DynamoDB', () => {
  test('returns a table', () => {
    const app = new App();
    const stack = new Stack(app, 'TestStack', { env: { account: '123456789', region: 'us-east-1' } });
    getTable(stack);
    const cfn = SynthUtils.toCloudFormation(stack);

    expect(cfn).toMatchSnapshot();
    expect(stack).toCountResources('AWS::DynamoDB::Table', 1);
  });
});
