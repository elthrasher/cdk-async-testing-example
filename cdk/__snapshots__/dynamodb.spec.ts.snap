// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamoDB returns a table 1`] = `
Object {
  "Resources": Object {
    "PaymentsTable48A42630": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "pk",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "pk",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "payments",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
